sda-commons-client-oidc/README.md

```diff
@@ -0,0 +1,95 @@
+# SDA Commons Client OIDC
+
+[![javadoc](https://javadoc.io/badge2/org.sdase.commons/sda-commons-client-oidc/javadoc.svg)](https://javadoc.io/doc/org.sdase.commons/sda-commons-client-oidc)
+
+The module `sda-commons-client-oidc` provides support for requesting OIDC access tokens from
+respective provides (e.g. Keycloak) for subsequent requests to other services.
+
+
+## Usage
+
+The [`OidcClient`](./src/main/java/org/sdase/commons/client/oidc/OidcClient.java) can be instantiated
+with a Jersey [`ClientFactory`](../sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/ClientFactory.java)
+and the respectable [`OidcConfiguration`](./src/main/java/org/sdase/commons/client/oidc/OidcConfiguration.java).
+The client also implements caching that is configured by the expiration property provided by the OIDC provider.
+
+```java
+OidcClient oidcClient = new OidcClient(jerseyClientBundle.getClientFactory(), configuration.getOidc());
+OidcResult oidcResult = oidcClient.createAccessToken();
+String accessToken = oidcResult.getAccessToken();
+```
+
+An easy way to integrate the OidcClient into existing PlatformClients is by creating a ClientRequestFilter
+and passing it to the PlatformClientBuilder:
+
+
+```java
+public class OidcRequestFilter implements ClientRequestFilter {
+
+  private OidcClient oidcClient;
+
+  public OidcRequestFilter(ClientFactory clientFactory, OidcConfiguration oidc) {
+      this.oidcClient = new OidcClient(clientFactory, oidc);
+  }
+
+  @Override
+  public void filter(ClientRequestContext requestContext) {
+    requestContext.getHeaders().add(AUTHORIZATION, "Bearer " + oidcResult.getAccessToken());
+  }
+}
+```
+
+```java
+ExternalServiceClient client = 
+    jerseyClientBundle
+      .getClientFactory()
+      .platformClient(configuration.getAuditableArchiveHttpClient())
+      .addFilter(oidcRequestFilter)
+      .api(ExternalServiceClient.class)
+      .atTarget(externalServiceUrl);
+```
+
+
+## Configuration
+
+The oidc client is configured in the `config.yaml` of the application.
+
+Example config for **production** to be used with environment variables of the cluster configuration:
+```yaml
+oidc:
+  disabled: ${OIDC_DISABLED:-false}
+  grantType: ${OIDC_GRANT_TYPE:-client_credentials}
+  clientId: ${OIDC_CLIENT_ID}
+  clientSecret: ${OIDC_CLIENT_SECRET}
+  username: ${OIDC_USERNAME}
+  password: ${OIDC_PASSWORD}
+  issuerUrl: ${OIDC_ISSUER_URL}
+```
+
+- _OIDC_DISABLED_
+    * Disable retrieving a new token completely. Not meant for production! 
+    * Example: `false`
+
+- _OIDC_GRANT_TYPE_
+    * Sets the OIDC grant type.
+    * Example: `client_credentials`
+
+- _OIDC_CLIENT_ID_
+    * The client id.
+     * Example: `client_id`
+- _OIDC_CLIENT_SECRET_
+    * The client secret.
+    * Example: `s3cr3t`
+     
+- _OIDC_USERNAME_
+    * A username. Only used for grant type 'password'.
+    * Example: `john`
+
+- _OIDC_PASSWORD_
+    * A password. Only used for grant type 'password'.
```

Comment:
I wonder what @wurstbrot thinks about support the `password` grant type

Reply:
Password Grant Type is not allowed in OAuth2/2.1.
I recommend to add a note, for example:

> deprecated: A password. Only used for grant type 'password' (must not be used in OAuth2.1).

Reply:
I marked it as deprecated in the documentation, I think we have some use cases where customers still rely on versions older than 2/2.1